.travis.yml

```diff
@@ -1,22 +1,70 @@
 language: bash
-sudo: required
-script: bash shellcheck && sudo bash buildall
-dist: xenial
+dist: focal
+virt: vm
+group: edge
+os: linux
 services:
   - docker
-before_install:
-  - docker version
-  # Fix for Ubuntu Xenial apt-daily.service triggering
-  # https://unix.stackexchange.com/questions/315502/how-to-disable-apt-daily-service-on-ubuntu-cloud-vm-image
-  - |
-    while sudo fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock >/dev/null 2>&1; do
-      sleep 1
-    done
-  - sudo apt-get -qq update
-  - sudo apt-get install -y debian-archive-keyring debootstrap shellcheck
-deploy:
-  provider: script
-  script: bash pushall
-  skip_cleanup: true
-  on:
-    branch: master
+
+.build_job: &build_job
+  stage: build
+  before_install:
+    - docker version
+    # Fix for Ubuntu Xenial apt-daily.service triggering
+    # https://unix.stackexchange.com/questions/315502/how-to-disable-apt-daily-service-on-ubuntu-cloud-vm-image
+    - |
+      while sudo fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock >/dev/null 2>&1; do
+        sleep 1
+      done
+    - sudo rm /usr/local/bin/jq
+  install:
+    - sudo make .installed-requirements
+  script: sudo bash buildone $DIST $PLATFORM
+  after_success:
+    - 'if [[ "$DIST" == "buster" ]] ; then sudo docker tag "bitnami/minideb:$DIST-$PLATFORM" "$BASENAME:latest-$PLATFORM" ; fi'
+    - 'if [[ "$TRAVIS_BRANCH" == "master" ]] ; then sudo bash pushone $DIST $PLATFORM ; fi'
```

Comment:
question: This is being done here as it's on a different machine and so it can't be pushed in the deploy stage?

Reply:
Yes, exactly. Since every image is built on a different machine and there is no way to pass a docker image as artifact to another stage, the only way is to push the built image on `after_success` after the build is finished.

Reply:
Correction: tar files created by `buildone` can maybe passed as artifact and pushed only in deploy stage.
Is there any limit to artifact files on travis?

Reply:
I do not know.

One thing is that we run tests on the images after importing the tarball to the base image, which requires starting a container from the image. If we want to test all arches then that would have to be done on each arch I believe, which makes this the better way to do that, even with the pushing.

The alternative would be pushing images without tags to some holding registry, then having an artifact that reported the id and then doing the final push in deploy when everything is OK.

I'm not sure what is better, but this seems acceptable to me.